src/main/java/redis/clients/jedis/util/SafeEncoder.java

```diff
@@ -27,7 +26,7 @@ private SafeEncoder(){
   public static byte[] encode(final String str) {
     try {
       if (str == null) {
-        throw new JedisDataException("value sent to redis cannot be null");
+        throw new JedisException("null value cannot be sent to redis");
```

Comment:
I agree with the change, but this bit is **not** backward compatible

Reply:
Nice catch.

Reply:
This method throws `JedisException` from catch. Check `throw new JedisException(e);`. Would it still be backward incompatible?

Reply:
But that's only catching `UnsupportedEncodingException` which could be thrown from `str.getBytes(Protocol.CHARSET)`, so doesn't cover **this** throw

Reply:
Hmm... Well, from a different perspective, **this** throws exception only when `str == null`, which, IMO, is highly unlikely. Would it be worth adding this in a non-major release rather than waiting for a major release?

Reply:
I agree, it's fairly low risk case, I think it's fine in a minor release, we should mention it in the changelog explicitly as technically it is a breaking change.